packages/network/src/workers/SyncWorker.ts

```diff
@@ -187,7 +187,7 @@ export class SyncWorker<C extends Components> implements DoWork<Input, NetworkEv
 
     let initialState = createCacheStore();
 
-    if (initialRecords) {
```

Comment:
<img width="669" alt="fix__resolved_a_minor_bug_in_the_empty_array_validation_check_by_0xrhsmt_·_Pull_Request__1089_·_latticexyz_mud" src="https://github.com/latticexyz/mud/assets/54972320/ff6caa24-8c9b-447b-acb1-89a94210b28a">

empty array is truthy.